src/core/gui/sidebar/Sidebar.cpp

```diff
@@ -152,20 +158,20 @@ void Sidebar::selectPageNr(size_t page, size_t pdfPage) {
 
 void Sidebar::setSelectedPage(size_t page) {
     this->visiblePage = nullptr;
-    this->currentPage = nullptr;
+    this->currentPage = this->pages.end();
 
     size_t i = 0;
-    for (AbstractSidebarPage* p: this->pages) {
+    for (auto p = this->pages.begin(); p != this->pages.end(); p++) {
         if (page == i) {
-            gtk_widget_show(p->getWidget());
-            gtk_toggle_tool_button_set_active(GTK_TOGGLE_TOOL_BUTTON(p->tabButton), true);
-            this->visiblePage = p->getWidget();
+            gtk_widget_show((*p)->getWidget());
+            gtk_toggle_tool_button_set_active(GTK_TOGGLE_TOOL_BUTTON((*p)->tabButton), true);
+            this->visiblePage = (*p)->getWidget();
             this->currentPage = p;
-            p->enableSidebar();
+            (*p)->enableSidebar();
         } else {
-            p->disableSidebar();
-            gtk_widget_hide(p->getWidget());
-            gtk_toggle_tool_button_set_active(GTK_TOGGLE_TOOL_BUTTON(p->tabButton), false);
+            (*p)->disableSidebar();
+            gtk_widget_hide((*p)->getWidget());
+            gtk_toggle_tool_button_set_active(GTK_TOGGLE_TOOL_BUTTON((*p)->tabButton), false);
         }
```

Comment:
This is inefficient, when using a vector/deque, we have 3 phases: disable all pages previous to **and** after `page`, activate page.

Reply:
Could you explain further what you mean by that? Do you refer to something like branch prediction? Also I don't get how this step is inefficient with a vector but when using a list it isn't.

Reply:
Yes, that's a branch prediction concern.
You are right, it's also inefficient with a std::list. 
But as we only have 4 pages in average, it does not matter (I just reviewed the loop, without checking the whole context).
Just leave it as it.